website/docs/r/virtual_machine_gallery_application_assignment.html.markdown

```diff
@@ -0,0 +1,123 @@
+---
+subcategory: "Compute"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_virtual_machine_gallery_application_assignment"
+description: |-
+  Manages a Virtual Machine Gallery Application Assignment.
+---
+
+# azurerm_virtual_machine_gallery_application_assignment
+
+Manages a Virtual Machine Gallery Application Assignment.
+
+~> **NOTE:** Gallery Application Assignment can be defined either directly on `azurerm_linux_virtual_machine` and `azurerm_windows_virtual_machine` resources, or using the `azurerm_virtual_machine_gallery_application_assignment` resource - but the two ways cannot be used together. If both are used against the same Virtual Machine, spurious changes will occur.
```

Comment:
This note should also suggest the use of `ignore_changes` on the virtual machine resource for the `gallery_application` block.

```suggestion
~> **Note** Gallery Application Assignments can be defined either directly on `azurerm_linux_virtual_machine` and `azurerm_windows_virtual_machine` resources, or using the `azurerm_virtual_machine_gallery_application_assignment` resource - but the two approaches cannot be used together. If both are used with the same Virtual Machine, spurious changes will occur. It's recommended to use `ignore_changes` for the `gallery_application` block on the associated virtual machine resources, to avoid a persistent diff when using this resource.
```